smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -44,6 +44,65 @@ contract MixinTransfer is
   // This is calculated as `keyPrice * transferFeeBasisPoints / BASIS_POINTS_DEN`.
   uint public transferFeeBasisPoints;
 
+  /**
+  * @notice Allows the key owner to share their key (parent key) by
+  * transferring a portion of the remaining time to a new key (child key).
+  * @param _to The recipient of the shared key
+  * @param _tokenId the key to share
+  * @param _timeShared The amount of time shared
+  */
+  function shareKey(
+    address _to,
+    uint _tokenId,
+    uint _timeShared
+  ) public
+    onlyIfAlive
+    onlyKeyOwnerOrApproved(_tokenId)
+  {
+    require(_to != address(0), 'INVALID_ADDRESS');
+    address keyOwner = ownerOf[_tokenId];
+    require(getHasValidKey(keyOwner), 'KEY_NOT_VALID');
+    Key storage fromKey = keyByOwner[keyOwner];
+    Key storage toKey = keyByOwner[_to];
+    uint iDTo = toKey.tokenId;
+    uint time;
+    // get the remaining time for the origin key
+    uint timeRemaining = fromKey.expirationTimestamp - block.timestamp;
+    // get the transfer fee
+    uint fee = getTransferFee(keyOwner, _timeShared);
+
+    // ensure that we don't try to share too much
+    if(_timeShared.add(fee) < timeRemaining) {
+      time = _timeShared;
+      uint timePlusFee = _timeShared.add(fee);
+      // deduct time from parent key, including transfer fee
+      _timeMachine(_tokenId, timePlusFee, false);
+    } else {
+      fee = getTransferFee(keyOwner, timeRemaining);
```

Comment:
Hmm this seems okay but looking at it again, maybe we could simplify by using a standard transfer here.  At this point it should be on-par with the logic there.  We could maybe do that later..

Reply:
Yeah, I had a similar thought about this. 
The one difference here is that shareKey allows the original owner to retain their original tokenId (for an expired key), but maybe that doesn't matter. Maybe something to look at after this release.